dask/callbacks.py

```diff
@@ -1,27 +1,61 @@
-from collections import namedtuple
-
 from .context import _globals
 
-Callback = namedtuple('Callback', ['start', 'pretask', 'posttask', 'finish'])
+__all__ = ['Callback', 'add_callbacks']
+
+
+class Callback(object):
+    """Base class for using the callback mechanism."""
+
+    def __init__(self, start=None, pretask=None, posttask=None, finish=None):
+        self._start = start
+        self._pretask = pretask
+        self._posttask = posttask
+        self._finish = finish
```

Comment:
I set them private, as the user should never directly call these. In general, I feel that things like:

```python
In[1]: prof = Profiler()
In[2]: prof.<TAB>
```
should only show things the user should interact with.